### ISSUE TITLE
Fetch user's information from UIDAI

### ISSUE THREAD

**Issue by ashwineaso** [nic-delhi/AarogyaSetu_Android#353]
**Current Solution:**
Users are currently being asked to provide their personal information by manually answering the questions provided. Though this gives the user, the option to not provide accurate information (making the app useless), it also opens up cases for errors.

Also, the user is not provided with the option to edit their personal information after they enter it.

**Suggested Solution:**
Give the user, the option to let the app, to fetch their personal information from UIDAI (information which is with a govt. institution anyway), by providing their aadhaar number and OTP.

**Comment by surajkamblesk** [comment]
Yes exactly. By doing so people will not feed in wrong information just for the sake of testing and the data gathered can be used for contact tracing of the persons family . and in some cases might also help to provide any government scheme such as helping the covid warriors who are infected . etc
possibilities are many more.

**Comment by CatalanCabbage** [comment]
I would like to not bring fetching data from UIDAI into the loop too.  
Besides, if you '**Give the user the option** to let the app fetch their personal information', then it does not solve the problems you stated ('...to not provide accurate information...' and '...opens up cases for errors') since they still have the option to do it, right?  

So are you suggesting it's enforced? Which is not ideal. It's better to let the user know exactly what sensitive PII they're feeding into the system.

**Comment by ashwineaso** [comment]
@CatalanCabbage : Yes. Suggesting to make that optional, was a way of stating it mildly. ideally, it should be enforced, not optional. Because what I suggested was an Aadhaar based authentication. It DOESN'T "pull" any sensitive data into the app or system. It would just fetch the basic personal information about the user, which is analogous to Google/ Facebook login. You aren't providing any more sensitive information than what you provide to any other app out there.

To all those who disagree with that argument - what is your biggest concern? Privacy? Bad user experience?
